### PR TITLE
mimic: cephfs-journal-tool: get and check layout info during dump/undump

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/24933

### CODE REVIEW

**Comment by smithfarm** [comment]
@batrick Conflict resolution

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-08-20-2118-mimic](https://trello.com/c/hWHRSkCU/378-wip-yuri3-testing-2018-08-20-2118-mimic)

**Comment by batrick** [comment]
@yuriw this broke your build

**Comment by yuriw** [comment]
@pdvian see https://shaman.ceph.com/builds/ceph/wip-yuri3-testing-2018-08-20-2118-mimic/
Pls add "needs-qa" and "13.2.2" what done

**Comment by smithfarm** [comment]
```
/build/ceph-13.2.1-95-gc131035/src/tools/cephfs/Dumper.cc: In member function 'int Dumper::dump(const char*)':
/build/ceph-13.2.1-95-gc131035/src/tools/cephfs/Dumper.cc:124:6: error: 'fsid_str' was not declared in this scope
      fsid_str,
      ^~~~~~~~
/build/ceph-13.2.1-95-gc131035/src/tools/cephfs/Dumper.cc:124:6: note: suggested alternative: 'fsid_t'
      fsid_str,
      ^~~~~~~~
      fsid_t
```

File: src/tools/cephfs/Dumper.cc [hunk 1]
```diff
@@ -108,13 +108,20 @@ int Dumper::dump(const char *dump_file)
...
 	    role.rank, 
 	    (unsigned long long)start, (unsigned long long)start,
 	    (unsigned long long)len, (unsigned long long)len,
 	    (unsigned long long)journaler.last_committed.write_pos, (unsigned long long)journaler.last_committed.write_pos,
 	    (unsigned long long)journaler.last_committed.stream_format,
 	    (unsigned long long)journaler.last_committed.trimmed_pos, (unsigned long long)journaler.last_committed.trimmed_pos,
+            (unsigned long)journaler.last_committed.layout.stripe_unit, (unsigned long)journaler.last_committed.layout.stripe_unit,
+            (unsigned long)journaler.last_committed.layout.stripe_count, (unsigned long)journaler.last_committed.layout.stripe_count,
+            (unsigned long)journaler.last_committed.layout.object_size, (unsigned long)journaler.last_committed.layout.object_size,
+	    fsid_str,
```
Comment (smithfarm) [inline src/tools/cephfs/Dumper.cc]: This line should be removed from the cherry-pick, as it is not part of 989ecf234a24c53884cf202de588692a17ccec2a

**Comment by smithfarm** [comment]
@pdvian Could you please remove the line as requested above?

**Comment by smithfarm** [comment]
@pdvian Ping?

**Comment by yuriw** [comment]
[wip-yuri4-testing-2018-10-15-0002-mimic](https://trello.com/c/hEUw3v8h/440-wip-yuri4-testing-2018-10-15-0002-mimic)

**Comment by smithfarm** [comment]
closing in favor of https://github.com/ceph/ceph/pull/24583

**Comment by smithfarm** [comment]
@yuriw This commit broke an earlier build and was not fixed. Did it break the most recent build?

In any case, I closed the PR now and opened a new one which is (hopefully) correct.

**Pull request closed without merging**